# bigaslab
Files and Scripts generated at Bigas lab
# Yolanda Guillén


## Description of scrips
ATACseq.R → Rbpj ATAC-Seq C.Ruiz
ArrayLimma_Script.R → Analysis of microarrays b-cat/NICD mouse C. Gekas
CEBPA_rnaseq.R → Bulk RNASeq CEBPA and WT1 AML samples P. Nomdedéu
ChIPSeq.R → b-catenin CHIPSeq initial analysis
ChIP_bcat_mice.R → Annotation of b-cat and partners in mouse and cell lines
ChIPhist_CKC.R → Brd4 ChIPSeq analysis
ChIPhist_IKB.R → Histone ChIPS Ikb Laura Marruecos
ChIPhist_PSuz → ChIPs Laura Marruecos
Clinical_TCell.R → T-ALL RNASeq cohorts for splicing analysis and database
Cytodata.R → Analysis of flow cytometry data with R
DEGs_signature_chemo → Chemoresistance and tumoroids Teresa Lobo
DESeq2.r → RNASeq data analysis from C. Porcheri Dll4 project
Dx_relapse_Dec2021.ipynb → scRNAseq paired samples Dx Rx from a T-ALL patient
GSEA.R → GSEA analysis from C.Porcheri RNASeq data
Gviz.R → Example script for Gviz library
RNASeq_HSC_IKB.R → Bulk RNASeq data IKB KO from HSC in mice R. Thambyrajah
RNASeq_fetal_Ruiz.R → Bulk RNASeq data RBPj KO and WT from LSK C. Ruiz
RNASeq_IKB → Bulk RNASeq data from Laura Marruecos EphB2 pheno
RNAseq_htseq_cuff.R → Bulk RNASeq data RBPj KO and public datasets C. Ruiz
RNAseq_tumoroids.R → Bulk RNASeq data fetal signature L. Marruecos
Stat_FG.R → Script for data STAT4 F. Gallardo
TALL_ASevent.R → Analysis of DE splicing events Splicing project
TCGA_work.R → Example script to get data from TCGA using TCGA biolinks
T_ALL.R → Microarrays expression from GEO BioEssays review
T_ALL_GEOquery.R → GEO Microarrays expression b-cat signature and survivals
T_ALL_Teresa_surv.R → Script by Teresa Lobo for survival curves
bcat_ASevent.R → b-cat regulated splicing events Splicing project
bcat_project_2021.R → b-cat transcription program in T-ALL. B-cat paper
cancer-app.R → shiny app to run with exec_shinny.R and Clinical_TCell.R
cancertool.R → Tumoroids and fetal signatures, T. Lobo
celegans.R → Analysis of C. elegans RNASeq and ChIPseq data
chips_examples.R → Exercises and examples ChIPSeq analysis
consensus_cluster_TALL.R → Classification in T-ALL subgroups random forest (use with Clinical_TCell.R)
exec_shinny.R → Database run with cancer-app and Clinical_TCell.R
foldchange_peaks.R → comparing foldchange peaks Histone mark in RPMI bcat
foldchange_peaks_Cr_Vi.R → Analysis of Crypta and Villus Cherry signaling
foldchange_peaks_IKK.R → ChIPSeq foldchange IKK KO and WT Brd4
htseq_selec_peaks_atac.R → From R sscript ATACseq.R extract and annotate peaks
isoforms.R → Splicing RNASeq Gekas bcat sh with library  isoformswitchanalyzeR
qPCR.R → Comparing Ct values from qPCR, Zara LEF1 isoforms qPCR
random_forest.R → classification RNASeq samples using random forest model
random_peaks.R → Statistical significance matching random and bed files peaks
rnaseq_Gekas_bcat.R → RNASeq analysis from cuffdiff pipeline b-cat Gekas
scRNAseq_AGM_2021.R → scRNASeq analysis AGM Roshana
scRNAseq_Dll4.R → scRNASeq analysis Dll4 experiment C. Porcheri
scRNAseq_Elaine.R → scRNASeq analysis Elaine data
scRNAseq_cam.R → original script scRNASeq analysis Cambridge 2019
selbal.R → J. Rivera selbal package, random forest classification
ssgsea-gui.R → local ssgsea running script, downloaded
supclust_ex.R → script random forest classification example dataset
tcR_clones.R → Analysis of TCR sequencing from mice transplant, F. Gallardo
