# bigaslab
Files and Scripts generated at Bigas lab (2018 - 2022)
### Yolanda Guillén January 2022


## Description of scrips
ATACseq.R → Rbpj ATAC-Seq C.Ruiz<br/>
ArrayLimma_Script.R → Analysis of microarrays b-cat/NICD mouse C. Gekas<br/>
CEBPA_rnaseq.R → Bulk RNASeq CEBPA and WT1 AML samples P. Nomdedéu<br/>
ChIPSeq.R → b-catenin CHIPSeq initial analysis<br/>
ChIP_bcat_mice.R → Annotation of b-cat and partners in mouse and cell lines<br/>
ChIPhist_CKC.R → Brd4 ChIPSeq analysis<br/>
ChIPhist_IKB.R → Histone ChIPS Ikb L. Marruecos<br/>
ChIPhist_PSuz → ChIPs Laura Marruecos<br/>
Clinical_TCell.R → T-ALL RNASeq cohorts for splicing analysis and database<br/>
Cytodata.R → Analysis of flow cytometry data with R<br/>
DEGs_signature_chemo → Chemoresistance and tumoroids Teresa Lobo<br/>
DESeq2.r → RNASeq data analysis from C. Porcheri Dll4 project<br/>
Dx_relapse_Dec2021.ipynb → scRNAseq paired samples Dx Rx from a T-ALL patient<br/>
genes_merge.R → Analysis of paired Dx and Rx scRNASeq from I. Sentís<br/>
GSEA.R → GSEA analysis from C.Porcheri RNASeq data<br/>
Gviz.R → Example script for Gviz library<br/>
RNASeq_HSC_IKB.R → Bulk RNASeq data IKB KO from HSC in mice R. Thambyrajah<br/>
RNASeq_fetal_Ruiz.R → Bulk RNASeq data RBPj KO and WT from LSK C. Ruiz<br/>
RNASeq_IKB → Bulk RNASeq data from L. Marruecos EphB2 pheno<br/>
RNAseq_htseq_cuff.R → Bulk RNASeq data RBPj KO and public datasets C. Ruiz<br/>
RNAseq_tumoroids.R → Bulk RNASeq data fetal signature L. Marruecos<br/>
Stat_FG.R → Script for data STAT4 F. Gallardo<br/>
scRNASeq_TenX_RBPj.ipynb --> scanpy notebook analysis of scRNASeq from RBPj KO and WT mice P. Herrero <br/>
TALL_ASevent.R → Analysis of DE splicing events Splicing project<br/>
TCGA_work.R → Example script to get data from TCGA using TCGA biolinks<br/>
T_ALL.R → Microarrays expression from GEO BioEssays review<br/>
T_ALL_GEOquery.R → GEO Microarrays expression b-cat signature and survivals<br/>
T_ALL_Teresa_surv.R → Script by T. Lobo for survival curves<br/>
bcat_ASevent.R → b-cat regulated splicing events Splicing project<br/>
bcat_project_2021.R → b-cat transcription program in T-ALL. B-cat paper<br/>
cancer-app.R → shiny app to run with exec_shinny.R and Clinical_TCell.R<br/>
cancertool.R → Tumoroids and fetal signatures, T. Lobo<br/>
celegans.R → Analysis of C. elegans RNASeq and ChIPseq data<br/>
chips_examples.R → Exercises and examples ChIPSeq analysis<br/>
consensus_cluster_TALL.R → Classification in T-ALL subgroups random forest (use with Clinical_TCell.R)<br/>
exec_shinny.R → Database run with cancer-app and Clinical_TCell.R<br/>
foldchange_peaks.R → comparing foldchange peaks Histone mark in RPMI bcat<br/>
foldchange_peaks_Cr_Vi.R → Analysis of Crypta and Villus Cherry signaling<br/>
foldchange_peaks_IKK.R → ChIPSeq foldchange IKK KO and WT Brd4<br/>
htseq_selec_peaks_atac.R → From R sscript ATACseq.R extract and annotate peaks<br/>
isoforms.R → Splicing RNASeq Gekas bcat sh with library  isoformswitchanalyzeR<br/>
qPCR.R → Comparing Ct values from qPCR, Zara LEF1 isoforms qPCR<br/>
random_forest.R → classification RNASeq samples using random forest model<br/>
random_peaks.R → Statistical significance matching random and bed files peaks<br/>
rnaseq_Gekas_bcat.R → RNASeq analysis from cuffdiff pipeline b-cat Gekas<br/>
scRNAseq_AGM_2021.R → scRNASeq analysis AGM Roshana<br/>
scRNAseq_Dll4.R → scRNASeq analysis Dll4 experiment C. Porcheri<br/>
scRNAseq_Elaine.R → scRNASeq analysis Elaine data<br/>
scRNAseq_cam.R → original script scRNASeq analysis Cambridge 2019<br/>
selbal.R → J. Rivera selbal package, random forest classification<br/>
ssgsea-gui.R → local ssgsea running script, downloaded<br/>
supclust_ex.R → script random forest classification example dataset<br/>
tcR_clones.R → Analysis of TCR sequencing from mice transplant, F. Gallardo<br/>
